esx_service/vmdk_ops.py

```diff
@@ -660,11 +660,16 @@ def listVMDK(tenant):
             for x in vmdks]
 
 
-# Return VM managed object, reconnect if needed. Throws if connection fails twice.
-# returns None if the uuid is not found
-def findVmByUuid(vm_uuid):
+
+def findVmByUuid(vm_uuid, is_vc_uuid=False):
+    """
+    Find VM by vm_uuid.
+    is_vc_uuid should be true if vm_uuid is vc uuid, else it should be false.
+    Return VM managed object, reconnect if needed. Throws if connection fails twice.
+    Returns None if the uuid is not found
+    """
     si = get_si()
-    vm = si.content.searchIndex.FindByUuid(None, vm_uuid, True, False)
+    vm = si.content.searchIndex.FindByUuid(None, vm_uuid, True, is_vc_uuid)
```

Comment:
Why not simply pass True first time and then if vm is None (or exception happens) then redo the call passing False.

All other callers don't have to change to pass the extra arg and other callers don't need to redo the call to this function passing the same UUID with a "False". FindVmByUUID _knows_ that if passing in True to si.content.searchIndex.FindByUuid doesn't work then call the same passing False.